javaslang/src/test/java/javaslang/collection/CollectionsTest.java

```diff
@@ -0,0 +1,68 @@
+/*     / \____  _    _  ____   ______  / \ ____  __    _______
+ *    /  /    \/ \  / \/    \ /  /\__\/  //    \/  \  //  /\__\   JΛVΛSLΛNG
+ *  _/  /  /\  \  \/  /  /\  \\__\\  \  //  /\  \ /\\/ \ /__\ \   Copyright 2014-2017 Javaslang, http://javaslang.io
+ * /___/\_/  \_/\____/\_/  \_/\__\/__/\__\_/  \_//  \__/\_____/   Licensed under the Apache License, Version 2.0
+ */
+package javaslang.collection;
+
+import javaslang.Tuple;
+import org.junit.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class CollectionsTest {
+
+    @Test
+    public void shouldBeEqualSets() throws Exception {
+        forAll(List.of(TreeSet.ofAll(1, 2, 3),
+                HashSet.ofAll(1, 2, 3),
+                LinkedHashSet.ofAll(1, 2, 3),
+                BitSet.ofAll(1, 2, 3)), true);
+    }
+
+    @Test
+    public void shouldBeEqualSeqs() throws Exception {
+        forAll(List.of(Array.ofAll(1, 2, 3),
+                Stream.ofAll(1, 2, 3),
+                Vector.ofAll(1, 2, 3),
+                List.ofAll(1, 2, 3),
+                Queue.ofAll(1, 2, 3)), true);
+    }
+
+    @Test
+    public void shouldBeEqualMaps() throws Exception {
+        forAll(List.of(TreeMap.of(1, 2, 2, 3, 3, 4),
+                HashMap.of(1, 2, 2, 3, 3, 4),
+                LinkedHashMap.of(1, 2, 2, 3, 3, 4)), true);
+    }
+
+    @Test
+    public void shouldBeEqualMultimaps() throws Exception {
+        forAll(List.of(TreeMultimap.withSeq().<Integer, Integer>empty().put(1, 1).put(1, 1).put(2, 2),
+                HashMultimap.withSeq().<Integer, Integer>empty().put(1, 1).put(1, 1).put(2, 2),
+                LinkedHashMultimap.withSeq().<Integer, Integer>empty().put(1, 1).put(1, 1).put(2, 2)), true);
+    }
+
+    @Test
+    public void shouldNotBeEqualSeqAndSet() throws Exception {
+        forAll(List.of(Array.ofAll(1, 2, 3),
+                TreeSet.ofAll(1, 2, 3)), false);
+    }
+
+    @Test
+    public void shouldNotBeEqualMapAndSet() throws Exception {
+        forAll(List.of(HashSet.of(Tuple.of(1, 2), Tuple.of(2, 3)),
+                TreeMap.of(1, 2, 2, 3)), false);
+    }
+
+    private void forAll(List<Traversable<?>> traversables, boolean value) {
```

Comment:
what do you think, does makes sense to create additional private method to check hashCode equality?